lib/dash/segment_template.js

```diff
@@ -275,9 +278,20 @@ shaka.dash.SegmentTemplate = class {
     // Capture values that could change as the parsing context moves on to
     // other parts of the manifest.
     const periodStart = context.periodInfo.start;
-    const periodDuration = context.periodInfo.duration;
-    const periodEnd = periodDuration ?
-        periodStart + periodDuration : Infinity;
+    const periodId = context.period.id;
+    const initialPeriodDuration = context.periodInfo.duration;
+
+    // For multi-period live streams the period duration may not be known until
+    // the following period appears in an updated manifest. periodDurationMap
+    // provides the updated period duration.
+    const getPeriodEnd = () => {
+      const periodDuration =
+        (periodDurationMap && periodId && periodDurationMap[periodId]) ||
```

Comment:
You can do away with the check for periodDurationMap, since you specified the type as non-nullable.  The compiler won't let anyone pass a null value.

Since periodId is a nullable string, the check for that _should_ be okay here, but this looks like the kind of thing that could bite us in the future if we ever move to a numerical ID system.  If periodId were a number and was 0, this boolean check on periodId would cause an issue.

Though we don't have any plans to change that type, since you're really making sure the period _has_ an ID, I think it would be best to check periodId != null here.

Reply:
Will do